src/Workspaces/Core/Portable/Formatting/Engine/AbstractFormatEngine.cs

```diff
@@ -169,7 +169,7 @@ protected virtual NodeOperations CreateNodeOperationTasks(CancellationToken canc
             {
                 using (Logger.LogBlock(FunctionId.Formatting_CollectIndentBlock, cancellationToken))
                 {
-                    return AddOperations<IndentBlockOperation>(task.Result, (l, n) => _formattingRules.AddIndentBlockOperations(l, n), cancellationToken);
+                    return AddOperations<IndentBlockOperation>(task.Result, (l, n) => _formattingRules.AddIndentBlockOperations(l, n, _token2), cancellationToken);
```

Comment:
Here's the starting point where we take the last token (already stored in the FormatEngine) and we pass it along so that it can reach the formatting rule.